hcloud/resource_hcloud_reversedns.go

```diff
@@ -0,0 +1,240 @@
+package hcloud
+
+import (
+	"context"
+	"fmt"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hetznercloud/hcloud-go/hcloud"
+	"log"
+	"net"
+	"strings"
+)
+
+func resourceReverseDns() *schema.Resource {
+	return &schema.Resource{
+		Read:   resourceReverseDnsRead,
+		Create: resourceReverseDnsCreate,
+		Delete: resourceReverseDnsDelete,
+
+		Schema: map[string]*schema.Schema{
+			"server_id": {
+				Type:     schema.TypeInt,
+				Optional: true,
+				ForceNew: true,
+			},
+			"floating_ip_id": {
+				Type:     schema.TypeInt,
+				Optional: true,
+				ForceNew: true,
+			},
+			"ip_address": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"dns_ptr": {
+				Type:     schema.TypeString,
+				Computed: true,
+				Optional: true,
```

Comment:
Why is the `dns_ptr` optional? The the code will break if it is not provided.

Reply:
It is optional because we need to make it possible to "change to the default" value

Reply:
When you want to change to the default value you have to delete the terraform resource, this will reset/delete the reverse dns pointer.
Using terraform for negative assertions for your infrastructure like: "make sure this dns ptr is NOT set" does not make much sense to me.